src/components/url-input.js

```diff
@@ -0,0 +1,90 @@
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Input, Tooltip } from 'antd';
+import React, { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useService } from './container-context.js';
+import { SOURCE_TYPE } from '../domain/constants.js';
+import PublicIcon from './icons/general/public-icon.js';
+import { getSourceType } from '../utils/media-utils.js';
+import ClientConfig from '../bootstrap/client-config.js';
+import PrivateIcon from './icons/general/private-icon.js';
+import ResourcePicker from './resource-picker/resource-picker.js';
+import { GlobalOutlined, WarningOutlined, YoutubeOutlined } from '@ant-design/icons';
+import { ensurePortableUrlIfStorageUrl, storageLocationPathToUrl } from '../utils/storage-utils.js';
+
+function UrlInput({ sourceUrl, onChange }) {
+  const { t } = useTranslation('urlInput');
+  const clientConfig = useService(ClientConfig);
+
+  const sourceType = useMemo(
+    () => getSourceType({ sourceUrl, cdnRootUrl: clientConfig.cdnRootUrl })
+    , [clientConfig.cdnRootUrl, sourceUrl]
+  );
+
+  const inputPrefixIcon = useMemo(() => {
+    switch (sourceType) {
+      case SOURCE_TYPE.none:
+        return null;
+      case SOURCE_TYPE.youtube:
+        return <YoutubeOutlined />;
+      case SOURCE_TYPE.internalPublic:
+        return <PublicIcon />;
+      case SOURCE_TYPE.internalPrivate:
+        return <PrivateIcon />;
+      case SOURCE_TYPE.external:
+        return <GlobalOutlined />;
+      default:
+        return <WarningOutlined />;
+    }
+  }, [sourceType]);
+
+  const handleUrlChange = selectedPortableUrl => {
+    onChange({ sourceUrl: selectedPortableUrl, sourceType });
+  };
+
+  const handleInputValueChange = event => {
+    onChange({ sourceUrl: event.target.value, sourceType });
```

Comment:
Reconsidered the return value, will return the url only and the type can be identified wherever needed. I overcomplicated it this way.